src/cascadia/TerminalSettingsEditor/Profiles.cpp

```diff
@@ -248,6 +245,59 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
             StartingDirectoryUseParentCheckbox().IsChecked(true);
         }
 
+        // Subscribe to some changes in the view model
+        // These changes should force us to update our own set of "Current<Setting>" members,
+        // and propagate those changes to the UI
+        _State.Profile().PropertyChanged([&](auto&&, const PropertyChangedEventArgs& args) {
```

Comment:
**NEVER CAPTURE `this` BY REFERENCE** you will absolutely crash :smile:

Reply:
you will want a revoker, otherwise the profile will get more and more and more event handlers for PropertyChanged accumulating on it as you navigate to and away from the page.

Reply:
I'm really new to revokers so I'm leaving this as unresolved just to confirm that I did it right.
- Introduced `INotifyPropertyChanged::PropertyChanged_revoker _ViewModelChangedRevoker;`
- updated this line to be...
```cpp
_ViewModelChangedRevoker = _State.Profile().PropertyChanged(winrt::auto_revoke, [weakThis{ get_weak() }](auto&&, const PropertyChangedEventArgs& args) {
```
- Added this:
```cpp
void Profiles::OnNavigatedFrom(const NavigationEventArgs& e)
{
    _ViewModelChangedRevoker.revoke();
}
```

My thought process is, we don't actually construct a new page on navigation. So the `auto_revoke` is really only useful when we close SUI. On navigation, we update the revoker and set the property changed handler. When we navigate away, we should stop listening to these because they're now outdated.

Reply:
The revoke triggers when you close SUI (the page gets destructed) or when it gets overwritten in OnNvaigatedTo. It's a bonus to put it in OnNavigatedFrom. If you do that, please add the other revokers for other event handlers established on OnNavTo :wink:

Reply:
We just listen to the view model's property changed event so I think that's it?